File: Day-03-Working with Arrays/08-Union-Arrays/Solution.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n;

    int arr1[n];
    for(int i = 0; i < n; i++)
        cin >> arr1[i];

    cin >> m;

    int arr2[m];
    for(int i = 0; i < m; i++)
        cin >> arr2[i];

    int i = 0, j = 0;

    while(i < n && j < m) {
        if(arr1[i] < arr2[j]) {
            if(i == 0 || arr1[i] != arr1[i-1])
                cout << arr1[i] << " ";
            i++;
        }
        else if(arr2[j] < arr1[i]) {
            if(j == 0 || arr2[j] != arr2[j-1])
                cout << arr2[j] << " ";
            j++;
        }
        else {
            if(i == 0 || arr1[i] != arr1[i-1])
                cout << arr1[i] << " ";
            i++;
            j++;
        }
    }

    while(i < n) {
        if(i == 0 || arr1[i] != arr1[i-1])
            cout << arr1[i] << " ";
        i++;
    }

    while(j < m) {
        if(j == 0 || arr2[j] != arr2[j-1])
            cout << arr2[j] << " ";
        j++;
    }
}

File: Day-03-Working with Arrays/08-Union-Arrays/Solution.py
n = int(input())
arr1 = list(map(int, input().split()))

m = int(input())
arr2 = list(map(int, input().split()))

i = j = 0
result = []

while i < n and j < m:
    if arr1[i] < arr2[j]:
        if i == 0 or arr1[i] != arr1[i-1]:
            result.append(arr1[i])
        i += 1
    elif arr2[j] < arr1[i]:
        if j == 0 or arr2[j] != arr2[j-1]:
            result.append(arr2[j])
        j += 1
    else:
        if i == 0 or arr1[i] != arr1[i-1]:
            result.append(arr1[i])
        i += 1
        j += 1

while i < n:
    if i == 0 or arr1[i] != arr1[i-1]:
        result.append(arr1[i])
    i += 1

while j < m:
    if j == 0 or arr2[j] != arr2[j-1]:
        result.append(arr2[j])
    j += 1

print(*result)